frontend-utils/src/backends/navigator.rs

```diff
@@ -271,7 +272,16 @@ impl<F: FutureSpawner, I: NavigatorInterface> NavigatorBackend for ExternalNavig
                 })?;
 
                 let url = response.url().to_string();
-
+                let text_encoding =
+                    if let Some(content_type) = response.headers().get("Content-Type") {
+                        if let Ok(content_type_str) = content_type.to_str() {
+                            get_encoding(content_type_str)
```

Comment:
would this work? To reduce the nesting
```rs
let text_encoding = response
    .headers()
    .get("Content-Type")
    .and_then(|header| header.to_str())
    .and_then(|header| get_encoding(content_type_str));
```

Reply:
Sure, I can do that

Reply:
That approach works if you add a `.ok()` after the `.to_str()`
Thanks for the suggestion :)